src/main/java/controller/Controller.java

```diff
@@ -0,0 +1,215 @@
+ */
+public class Controller {
+
+    //DAO's
+    PlayerDAO playerDAO;
+    AreaDAO areaDAO;
+    EnemyDAO enemyDAO;
+    WeaponDAO weaponDAO;
+    TreasureDAO treasureDAO;
+
+
+    Map<Integer, Enemy> enemies;
+    Area area;
+    Player currentPlayer;
+
+    public Controller(){
+        playerDAO = new PlayerDAO();
+        areaDAO = new AreaDAO();
+        enemyDAO = new EnemyDAO();
+        weaponDAO = new WeaponDAO();
+        treasureDAO = new TreasureDAO();
+    }
+
+    public void setCurrentPlayer(String name){
+        currentPlayer = playerDAO.getPlayer(name);
+        area = currentPlayer.currentArea;
+        populateEnemies();
+    }
+
+    public void setCurrentArea(Area a){
+        area = a;
+        currentPlayer.currentArea = a;
+        populateEnemies();
+    }
+
+    public void createNewPlayer(String name){
+        currentPlayer = playerDAO.createPlayer(name);
+        area = currentPlayer.currentArea;
+        populateEnemies();
+    }
+
+    //VIEW USE THESE METHODS TO GET INFO YOU NEED
+
+    /**
+     * Get a list of all the player names in the DB
+     *
+     * @return List of all player names
+     */
+    public List<String> getAllPlayers(){
+        return playerDAO.getNamesOfAllPlayers();
+    }
+
+    /**
+     * Get a Map of all the enemies names in the current Area
+     *
+     *
+     *
+     * @return List of all enemies names
+     */
+    public Map<Integer, String> getAllEnemies(){
+        Map<Integer, String> enemyNames = new HashMap<>();
+        enemies.forEach((id, enemy) -> {
+            enemyNames.put(id, enemy.name);
+        });
+        return enemyNames;
+    }
+
+    /**
+     * Get the current player's name
+     *
+     * @return String of player name
+     */
+    public String getCurrentPlayerName(){
+        return currentPlayer.name;
+    }
+
+    /**
+     * Get the current weapon the current player is using
+     *
+     * @return the player's weapon
+     */
+    public String getCurrentWeapon(){
+        return currentPlayer.weapon.name;
+    }
+
+    /**
+     * Get all the abilities of a player
+     * @return
+     */
+    public Set<String> getAbilities(){
+        return currentPlayer.abilities.keySet();
+    }
+
+    /**
+     * Get 4 possible areas for when a player is travelling to a new room
+     *
+     * @return a list of the possible areas
+     */
+    public List<Area> getPossibleAreas(){
+        return AreaDAO.getAreasInLevelRange(getPlayerLevel(), 4);
+    }
+
+    public String getTreasure(){
+        return area.treasure.name;
+    }
+
+
+
+    //VIEW USE THESE METHODS TO COMPLETE A USER ACTION
+
+    /**
+     * Combat phase of player, attacking an enemy with their weapon
+     *
+     * After the damage is calculated to the enemy, all the enemies
+     * in the area will then attack the player
+     *
+     * @param enemyId The target enemy
+     */
+    public void Attack(Integer enemyId){
+        enemies.forEach((id, enemy) -> {
```

Comment:
I need to iterate over them anyway because they will each be attacking the player